src/libsyncengine/requests/serverrequests.cpp

```diff
@@ -1067,63 +1067,41 @@ ExitCode ServerRequests::getPublicLinkUrl(int driveDbId, const QString &fileId,
         return ExitCode::DataError;
     }
 
-    if (auto exitCode = job->runSynchronously(); exitCode != ExitCode::Ok) {
+    if (job->runSynchronously() != ExitCode::Ok) {
         if (job->exitCode() == ExitCode::BackError && job->exitCause() == ExitCause::ShareLinkAlreadyExist) {
-            // Get link
-            std::shared_ptr<GetFileLinkJob> job2;
+            // The link already exist, get it
```

Comment:
```suggestion
            // The link already exists, get it
```

Reply:
Done in 3479496b5a2452349bf511bb620f292b9214faba